shared/constants/chat2/meta.tsx

```diff
@@ -149,25 +166,12 @@ export const updateMeta = (
       (newMeta.trustedState === 'trusted' && oldMeta.trustedState !== 'trusted') ||
       newMeta.inboxLocalVersion > oldMeta.inboxLocalVersion
     ) {
-      const merged = {...newMeta}
-      if (shallowEqual([...merged.rekeyers], [...oldMeta.rekeyers])) {
-        merged.rekeyers = oldMeta.rekeyers
-      }
-      if (shallowEqual([...merged.resetParticipants], [...oldMeta.resetParticipants])) {
-        merged.resetParticipants = oldMeta.resetParticipants
-      }
-      if (shallowEqual(merged.retentionPolicy, oldMeta.retentionPolicy)) {
-        merged.retentionPolicy = oldMeta.retentionPolicy
-      }
-      if (shallowEqual(merged.teamRetentionPolicy, oldMeta.teamRetentionPolicy)) {
-        merged.teamRetentionPolicy = oldMeta.teamRetentionPolicy
-      }
-      return merged
+      return copyOverOldValuesIfEqual(oldMeta, newMeta)
     }
     return oldMeta
   }
   // higher inbox version, use new
-  return newMeta
+  return copyOverOldValuesIfEqual(oldMeta, newMeta)
```

Comment:
previously if we got a newer meta we just used the whole thing and if we had a untrusted to trusted flow we'd try and reuse previously equal values. instead we always try and reuse old values (most of them dont actually change). this stops things like the `retentionPolicy` struct from seemingly changing all the time